python/cudf/cudf/core/index.py

```diff
@@ -451,6 +451,27 @@ def isin(self, values):
 
         return result
 
+    def where(self, cond, other=None):
+        """
+        Replace values where the condition is False.
+
+        Parameters
+        ----------
+        cond : bool array-like with the same length as self
+            Where cond is True, keep the original value.
+            Where False, replace with corresponding value from other.
+            Callables are not supported.
+        other: scalar, or array-like
+            Entries where cond is False are replaced with
+            corresponding value from other. Callables are not
+            supported. Default is None.
+
+        Returns
+        -------
+        Same type as caller
+        """
+        return super().where(cond=cond, other=other)
```

Comment:
Can we not just inherit from the Frame implementation here?

Reply:
Thought so initially, but later ran into docs of `Index.where` and realized their function params don't support `inplace` so had to fork out a new signature in `Index` and call `Frame.where`.

https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Index.where.html

Reply:
Any reason for us not to support `inplace` on `Index`?

Reply:
Sure we can support it, do you want it to be added back?

Reply:
Got it, lets just leave it as is for now. Thanks for the explanation.